cpp/src/arrow/dataset/file_ipc.cc

```diff
@@ -151,6 +171,13 @@ Result<RecordBatchGenerator> IpcFileFormat::ScanBatchesAsync(
       ARROW_ASSIGN_OR_RAISE(generator, reader->GetRecordBatchGenerator(
                                            /*coalesce=*/false, options->io_context));
     }
+#ifdef ARROW_WITH_OPENTELEMETRY
+    opentelemetry::trace::StartSpanOptions span_options;
+    span_options.parent = parent_span->GetContext();
+    generator = arrow::internal::tracing::WrapAsyncGenerator(
```

Comment:
Why do I need to manually specify the parent here?  Can we add a comment explaining?

Reply:
It seems to be unnecessary - I've removed it